File: 0169-majority-element/0169-majority-element.py
class Solution:
    def majorityElement(self, nums: List[int]) -> int:
        hash = {}
        a = major = 0

        for n in nums:
            hash[n] = 1 + hash.get(n, 0)

            if hash[n] > major:
                a = n
                major = hash[n]

        return a

File: 0169-majority-element/0169-majority-element.cpp
class Solution {
public:
    int majorityElement(vector<int>& nums) {
        int count = 0, canditate = 0;
        for(int num : nums){
            if(count == 0){
                canditate = num;
            }
            if(canditate == num){
                count++;
            }else{
                count--;
            }
        }
        return canditate;
    }
};